index.html

```diff
@@ -2324,6 +2324,52 @@ <h3>Presentations</h3>
 [[?VC-JOSE-COSE]] specification.
         </p>
 
+        <section>
+          <h4>Enveloped Verifiable Credentials</h4>
+
+          <p>
+It is possible to include one or more <a>verifiable credentials</a> in a
+<a>verifiable presentation</a> that have been secured using a securing mechanism
+that "envelopes" the payload, such as [[[?VC-JOSE-COSE]]] [[?VC-JOSE-COSE]].
+This can be accomplished by associating the `verifiableCredential` property with
+an object that has a `type` of `EnvelopedVerifiableCredential`.
+          </p>
+
+          <dl>
+            <dt id="defn-EnvelopedVerifiableCredential">EnvelopedVerifiableCredential</dt>
+            <dd>
+Used to associate an object containing an enveloped <a>verifiable credential</a>
+with the `verifiableCredential` property in a <a>verifiable presentation</a>.
+The `@context` value of the object MUST follow all of the "MUST" statements in
+Section <a href="#contexts"></a>. The `id` value of the object MUST be a `data:`
```

Comment:
```suggestion
The possible `@context` value of the object MUST follow all of the "MUST" statements in
Section <a href="#contexts"></a>. The `id` value of the object MUST be a `data:`
```

Is it really _necessary_ to include a `@context` at this point? The only reason to have it is to use the `id` alias, but that would be inherited from the contexts above, wouldn't it?

Reply:
Yes, it's necessary, because we clear all context values inside the Verifiable Credential Graph. We have to do this in order to ensure that VCDM v1.1 VCs can be put inside VCDMV v2.0 VPs. If we don't do that, JSON-LD safe mode kicks in and alerts us that we're re-defining some terms (which is the correct behaviour).

Reply:
Ah. Right, my bad.

(For other reviewers, in case they do not follow: there is a `"@context": null` statement within the definition of the `VerifiablePresentation.verifiableCredential` property in the V2 `@context` file. The reason is what @msporny describes above.)

Let me picky, though. The text says:

> The @context value of the object MUST follow all of the "MUST" statements in Section [4.2 Contexts](https://pr-preview.s3.amazonaws.com/w3c/vc-data-model/pull/1379.html#contexts).

However, §4.2 only says:

> Verifiable credentials and verifiable presentations MUST include a @context [property](https://pr-preview.s3.amazonaws.com/w3c/vc-data-model/pull/1379.html#dfn-property).

Technically, i.e., in the vocabulary, an `EnvelopedVerifiableCredential` is _not_ a `VerifiableCredential`, meaning that the statement in §4.2 does not apply, i.e., it is not said that a `@context` file must be present. You can either change the text in §4.2, or add a statement in the new section whereby a `@context` property MUST be added.

Also, continuing to be picky, and looking at §5.12, the text says:

> For presentations, each value associated with the verifiableCredential property of the presentation is a separate named graph of type VerifiableCredentialGraph ***which contains a single verifiable credential.***

(Emphasis is mine.) I believe it should say something like

> which contains a single verifiable credential or and enveloped verifiable credential.

(Or something for that effect.)

Reply:
Yes, I believe your statements above are correct, I'll make those modifications in the next iteration of this PR.

Reply:
I don't think the second context is required here... if the v2 context defines things correctly.

Reply:
@OR13,

As explained in the comments above, the second context is indeed necessary as the context is intentionally cleared to support other contexts (such as 1.1). So a context declaration is required.

Reply:
@OR13 

We can put it another way: if the second context is not used, then the right terms for that object must be `@type` and `@id` instead of `type` and `id`, respectively (this is how the current, v2 version of `@context` works, and I do not see any way to avoid that.)

It is indeed unfortunate that the usage of those aliases force us to use the second context here. But that is water under the bridge, we decided not to use the original, `@`-prefixed values.

Reply:
@iherman, I applied your requests in 2ffb884853a93cc1195a114267ce25b31ff157e7.